module/VuFind/src/VuFind/Http/PhpEnvironment/Request.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * HTTP Request class
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2019.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  OAI_Server
```

Comment:
I think this package name should be changed. :-)

Reply:
Oops. I changed it to HTTP, but feel free to suggest better alternatives.

Reply:
I think it'll be obvious that the delegator doesn't work when all the command line tools start to fail, but yes, needs to be taken care of.